tests/experimental.libsonnet

```diff
@@ -96,33 +93,49 @@ local volumes = import 'templates/volumes.libsonnet';
               gcloud alpha compute tpus tpu-vm delete -q ${tpu_name} --zone=${zone}
               " > /scripts/cleanup.sh
 
-              echo "xl-ml-test:$(cat /scripts/id_rsa.pub)" > ssh-keys.txt
-              echo %(startupScript)s > startup-script.txt
-
-              # Retry every 30 seconds for 10 minutes
-              for i in {1..20}; do
-                set +e
-                gcloud alpha compute tpus tpu-vm create ${tpu_name} \
-                  --accelerator-type=%(acceleratorName)s \
-                  --version=%(softwareVersion)s  \
-                  --metadata-from-file='ssh-keys=ssh-keys.txt,startup-script=startup-script.txt' \
-                  --labels='test-name=%(testName)s' \
-                  --zone=${zone}
-
-                exit_code=$?
-                set -e
-                test $exit_code = 0 && break || sleep 30;
+              curl -X POST \
+                -H "Authorization: Bearer $(gcloud auth print-access-token)" \
+                -H "Content-Type: application/json" \
+                -d "{
+                  accelerator_type: %(acceleratorName)s,
+                  runtime_version: %(softwareVersion)s,
+                  network_config: {enable_external_ips: true},
+                  boot_disk: {source_image: 'projects/cloud-tpu-v2-images-dev/global/images/family/tpu-vm-tf-nightly'},
```

Comment:
Please move this `curl` logic into `tensorflow/`'s `common.TpuVm`. This change will affect all frameworks. We should prefer to use `gcloud` where we can, and only use `curl` where we need to set the boot disk override.

Reply:
Yeah I added that in the latest commit.